contracts/interfaces/IOriginationControllerSTIRFRY.sol

```diff
@@ -0,0 +1,31 @@
+// SPDX-License-Identifier: MIT
+
+pragma solidity 0.8.18;
+
+import "./IOriginationControllerBase.sol";
+
+import "../libraries/LoanLibrary.sol";
+
+interface IOriginationControllerSTIRFRY is IOriginationControllerBase {
+
+    // ============= Data Types =============
+
+    struct StirfryData {
+        address vaultedCurrency;
+        uint256 lenderVaultedCurrencyAmount;
+        uint256 borrowerVaultedCurrencyAmount;
+        uint256 payableToVaultedCurrencyRatio;
```

Comment:
Re: ratio here? Does this only imply a ratio in terms of the number of decimals each token uses, or does it also imply some "price ratio"? If it's decimals, should it be encoded in `allowedPairs` instead of being provided by a counterparty?

If it's provided by a counterparty, then it seems like the caller can put in whatever they want. If I were swapping variable-to-fixed, then I'd just make it `0` (it's also not validated anywhere, so I could). _Then,_ line 200 would mean that `stirfryData.borrowerVaultedCurrencyAmount` would have to equal 0. That's great for me, I just opened a loan without having to actually vault anything.

Reply:
Yup, the `payableToVaultedCurrencyRatio` is the decimals ratio. This ratio is currently being captured by the `allowedPairs` hash. See the hash setter function as well as the validation function.